libsolidity/ExpressionCompiler.cpp

```diff
@@ -818,6 +816,25 @@ void ExpressionCompiler::LValue::retrieveValue(Expression const& _expression, bo
 	}
 }
 
+void ExpressionCompiler::LValue::retrieveValueFromStorage(std::shared_ptr<Type const> const& _type, bool _remove) const
```

Comment:
Hm, these are deep internals of the LValue class, they should not be callable from outside.
I think it was a bad idea to extract _type from expression. Can you separate the two, make the expression optional (it is only used for error reporting) and only call `retrieveValue()` from outside?